File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n=nums.size();
        int left =0;
        int right=n-1;
        int index=n-1;
        vector<int>ans(n);
        while(left<=right){
            int leftsquare=nums[left]*nums[left];
            int rightsquare=nums[right]*nums[right];
            if(leftsquare>rightsquare){
                ans[index]=leftsquare;
                left++;
            }
            else{
                ans[index]=rightsquare;
                right--;
            }
            index--;

        }
        return ans;
    }
};

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int left=0;
        int right=nums.length-1;
        int index=nums.length-1;
        int [] ans=new int[nums.length];
        while(left<=right){
            int leftsq=nums[left]*nums[left];
            int rightsq=nums[right]*nums[right];

            if(leftsq<rightsq){
                ans[index]=rightsq;
                right--;
            }
            else{
                ans[index]=leftsq;
                left++;
            }
            index--;
        }
        return ans;
    }
}